web-registry/src/lib/ecocredit/api.ts

```diff
@@ -964,51 +962,33 @@ export const queryEcoBatches = async (
 ): Promise<BatchInfo[]> => {
   const client = await getQueryClient();
   try {
-    // Currently, the experimental testnet Hambach is running an old version of regen-ledger (v2.0.0-beta1)
-    // which provides a different API than latest v4.0 for querying credit batches.
-    if (expLedger) {
-      if (creditClassId) {
-        if (!params) {
-          params = new URLSearchParams();
-        }
-        params.set('class_id', creditClassId);
-      }
-      const { data } = await axios.get(
-        `${ledgerRESTUri}/regen/ecocredit/v1alpha1/batches`,
-        {
-          params,
-        },
+    // With regen-ledger v4.0, we first have to query all classes and then batches per class.
```

Comment:
I think we can get rid of the v4.0 logic since all networks have been upgraded to v5.0

Reply:
This logic is actually still used so we can't remove it for now as it involves larger refactoring.

Reply:
oh I see, this is actually an optimization that we have missed with 5.0 upgrade since we can now just query all batches in just one query if no credit class id provided

Reply:
I've created the task for fixing that: https://app.zenhub.com/workspaces/regen-registry-5f8998bec8958d000f4609e2/issues/gh/regen-network/regen-registry/1549